flang-rt/lib/runtime/pseudo-unit.cpp

```diff
@@ -124,15 +127,11 @@ std::size_t PseudoOpenFile::Read(
 
 std::size_t PseudoOpenFile::Write(FileOffset at, const char *buffer,
     std::size_t bytes, IoErrorHandler &handler) {
-  if (at) {
-    handler.Crash("%s: unsupported", RT_PRETTY_FUNCTION);
-  }
-  // TODO: use persistent string buffer that can be reallocated
-  // as needed, and only freed at destruction of *this.
-  auto string{SizedNew<char>{handler}(bytes + 1)};
-  runtime::memcpy(string.get(), buffer, bytes);
-  string.get()[bytes] = '\0';
-  std::printf("%s", string.get());
+  char *mutableBuffer{const_cast<char *>(buffer)};
+  char save{buffer[bytes]};
+  mutableBuffer[bytes] = '\0';
+  std::printf("%s", buffer);
+  mutableBuffer[bytes] = save;
```

Comment:
Just wondering why we need to care about this byte?

Reply:
That is clear to me :) but why do we need to restore the original byte value, if we always allocate an extra byte?

Reply:
It's a circular buffer.

Reply:
Ah! Thanks!